src/gtfs/read.rs

```diff
@@ -584,6 +604,34 @@ fn manage_comment_from_stop(
     comment_links
 }
 
+fn manage_odt_comment_from_stop_time(
+    comments: &mut CollectionWithId<objects::Comment>,
+    companies: &mut CollectionWithId<objects::Company>,
+    on_demand_transport_comment: Option<String>,
+    company_id: &str,
+) -> Option<String> {
+    if let Some(message) = on_demand_transport_comment {
+        if let Some(company) = companies.get(company_id) {
+            let comment_id = "ODT:".to_string() + &company.id;
+            let comment = comments.get(&comment_id);
+            if comment.is_none() {
+                let comment = objects::Comment {
+                    id: comment_id.clone(),
+                    comment_type: objects::CommentType::OnDemandTransport,
+                    label: None,
+                    name: message
+                        .replace("{agency_name}", &company.name)
+                        .replace("{agency_phone}", &company.phone.clone().unwrap_or_default()),
+                    url: None,
+                };
+                comments.push(comment).unwrap();
+            }
+            return Some(comment_id);
```

Comment:
You never need the `comment_id` (`String`), what you really need is the `Idx<Comment>`. So I believe you could modify this function to return `Option<Idx<Comment>>`. And 2 lines above, instead of doing `comments.push(comment).unwrap()` gives you the `Idx<Comment>` you're looking for.